### PR TITLE
Intersection Observer API

### PR DESCRIPTION
**Pull request by dutchenkoOleg**
feature detecting code is taken from official polyfill
https://github.com/w3c/IntersectionObserver/blob/master/polyfill/intersection-observer.js

### CODE REVIEW

**Comment by niksy** [comment]
1. Why `package-lock.json` changes?
2. This change also polyfills Intersection Observer. Modernizr should polyfill anything if not explicitly requested.

**Comment by dutchenkoOleg** [comment]
Changes added

File: feature-detects/dom/intersection-observer.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  }]
+}
+!*/
+/* DOC
+
+ Determines if Intersection Observer API is available.
+
+ */
+define(['Modernizr'], function (Modernizr) {
+  Modernizr.addTest('intersectionobserver', ‘IntersectionObserver’ in window);
```
Comment (rejas) [inline feature-detects/dom/intersection-observer.js]: only one type of ' please :-) those curly ‘ arent our type :-)

File: feature-detects/dom/intersection-observer.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  }, {
+    "name": "MDN documentation",
+    "href": "https://developer.mozilla.org/en/docs/Web/API/Intersection_Observer_API"
+  }]
+}
+!*/
+/* DOC
+
+ Determines if Intersection Observer API is available.
+
```
Comment (rejas) [inline feature-detects/dom/intersection-observer.js]: no empty lines needed

**Comment by dutchenkoOleg** [comment]
@rejas fixed :-)

**Comment by rejas** [comment]
Closes https://github.com/Modernizr/Modernizr/issues/2331

**Pull request merged**